### ISSUE TITLE
CPUID and RTDSC

### ISSUE THREAD

**Issue by xSanx** [HyperDbg/HyperDbg#58]
I can set a trap on CPUID and on RTDSC but how to read the registers immediately after calling these commands?

**Comment by SinaKarvandi** [comment]
If you're in debugger mode, you can use the 'r' command to view or modify registers. If you're on VMI Mode, you can use script engine, e.g, :

`printf("rax is : %llx", @rax);
`
if you want to modify a register from script engine then use the following script :
`@rax = 0x1234;`

**Comment by xSanx** [comment]
yes, but I can count  printf("rax is : %llx", @rax), but I'm interested to see what the processor wrote to the registers after the call CPUID or RTDSC

**Comment by SinaKarvandi** [comment]
Well, I think it will be triggered after the processor wrote the register, please take a look at:
https://github.com/HyperDbg/HyperDbg/blob/7838db39320da34622e854763d200805c1d082f0/hyperdbg/hprdbghv/code/vmm/vmx/HypervisorRoutines.c#L285

You can move the above line wherever you want and compile HyperDbg. I mean you can move it both after & before writing into RAX.

**Comment by SinaKarvandi** [comment]
By the way, I tag this issue as a feature to have pre- and post- events in the future.

**Comment by xSanx** [comment]
this project is still very cool

**Comment by SinaKarvandi** [comment]
By the way, I double-checked CPUID. If you access the registers then these registers are the registers that are already modified by the CPUID. Of course, the RAX is also changed but if you see the documentation !cpuid command, $context is the RAX before executing CPUID instruction. So, you can do whatever you want with your $context which is not modified RAX. As the RAX is modified when you handle it by using @rax.

Did you get it? Please let me know if it needs more clarification or explanation.
